src/library/scala/collection/IndexedSeqView.scala

```diff
@@ -37,6 +38,28 @@ object IndexedSeqView {
         current += 1
         r
       } else Iterator.empty.next()
+
+    override def drop(n: Int): Iterator[A] = {
+      if (n > 0) current = Math.min(self.size, current + n)
```

Comment:
same here for `current + n`

(and, FTR, the reverse Iterators look correct to me)